pkg/sriov/sriov.go

```diff
@@ -247,6 +247,17 @@ func (s *sriovManager) ApplyVFConfig(conf *types.NetConf) error {
 		if err := s.setVfGUID(conf, pfLink, conf.GUID); err != nil {
 			return err
 		}
+	} else {
+		// VF might already have guid. Verify if it is valid
+		vfLink, err := s.nLink.LinkByName(conf.HostIFNames)
+		if err != nil {
+			return fmt.Errorf("failed to lookup vf %q: %v", conf.HostIFNames, err)
+		}
+
+		hwAddr := vfLink.Attrs().HardwareAddr.String()
+		if hwAddr == "" || !utils.IsValidGUID(hwAddr[36:]) {
```

Comment:
perhaps add method `GUIDfromGID` or `GUIDFromHardwareAddr` so the `[36:]` is less wierd.

secondly we already know the format is OK we just care about the value (not zero and not fff....)
so why change to string just to run a regex on it ensuring its valid format since HardwareAddr.String() does just that - translate the bunch of bytes into a valid hardware address formatted string.

Reply:
in addition IsValidGUID does not check if guid is 0xffffffffffffffff